src/Microsoft.AspNet.Mvc.Localization/HtmlLocalizer.cs

```diff
@@ -0,0 +1,415 @@
+            new HtmlLocalizer(_localizer.WithCulture(culture), _encoder);
+
+        /// <summary>
+        /// Creates a new <see cref="LocalizedHtmlString"/> for a <see cref="LocalizedString"/>.
+        /// </summary>
+        /// <param name="result">The <see cref="LocalizedString"/>.</param>
+        protected LocalizedHtmlString ToHtmlString(LocalizedString result) =>
+            new LocalizedHtmlString(result.Name, result.Value, result.ResourceNotFound);
+
+        /// <summary>
+        /// Creates a <see cref="IStringLocalizer"/>.
+        /// </summary>
+        /// <param name="baseName">The base name of the resource to load strings from.</param>
+        /// <param name="location">The location to load resources from.</param>
+        /// <returns>The <see cref="IStringLocalizer"/>.</returns>
+        protected void CreateStringLocalizer(string baseName, string location) =>
+            _localizer = _localizerFactory.Create(baseName, location);
+
+        /// <summary>
+        /// Creates a <see cref="IStringLocalizer"/> using the <see cref="System.Reflection.Assembly"/> and
+        /// <see cref="Type.FullName"/> of the specified <see cref="Type"/>.
+        /// </summary>
+        /// <param name="type">The <see cref="Type"/>.</param>
+        protected void CreateStringLocalizer(Type type) => _localizer = _localizerFactory.Create(type);
+
+        /// <summary>
+        /// Encodes the arguments based on the object type.
+        /// </summary>
+        /// <param name="resourceString">The resourceString whose arguments need to be encoded.</param>
+        /// <param name="arguments">The array of objects to encode.</param>
+        /// <returns>The string with encoded arguments.</returns>
+        protected string EncodeArguments([NotNull] string resourceString, [NotNull] object[] arguments)
+        {
+            var position = 0;
+            var length = resourceString.Length;
+            var ch = '\x0';
+            StringBuilder builder = null;
+            var resourceStringBuilder = new StringBuilder();
+
+            while (true)
+            {
+                builder = new StringBuilder();
+
+                var isToken = false;
+                var p = position;
+                var i = position;
+
+                while (position < length)
+                {
+                    ch = resourceString[position];
+
+                    position++;
+                    if (ch == '}')
+                    {
+                        if (position < length && resourceString[position] == '}') // Treat as escape character for }}
+                        {
+                            position++;
+                            if (position < length && resourceString[position] != '{')
+                            {
+                                isToken = false;
+                                builder.Append(ch);
+                                resourceStringBuilder.Append(EncodeArgument(builder.ToString()));
```

Comment:
At this point `builder` contains characters copied directly from the resource. Thought the decision was to only encode non-`HtmlString` arguments?